### ISSUE TITLE
tomoScan.template and tomoScan_settings.req include both required and optional PVs

### ISSUE THREAD

**Issue by MarkRivers** [tomography/tomoscan#16]
tomoScan.template and tomoScan_settings.req include both required and optional PVs.  The "optional" PVs are actually beamline dependent.  Having both in the same files makes it more difficult to port to a new beamline.

I have now split tomoScan.template in 2 files:
  - tomoScan.template now contains only the files required by the tomoscan.py base class.
  - tomoScan_13BM.template contains the optional PVs that will be beamline-specific.  It may include PVs used in the derived class Python code,  and also metadata PVs that should be saved in the configuration file.

I have done the same for tomoScan_settings.req, splitting it into tomoScan_settings.req and tomoScan_13BM_settings.req.

I have modified the tomoScan.substitutions file to load both databases, and the auto_settings.req file to load both .req files.

I have tested the IOC and it runs identically as before the split.

I will now update the documentation to reflect this splitting of files.

**Comment by MarkRivers** [comment]
Documentation changes made:
- Renamed demo.rst to example_application.rst which is a better description.
- Added descriptions of both databases to example_application.rst
- Deleted screenshots.rst, moved that into example_application.rst

**Comment by MarkRivers** [comment]
Things that still need to be done:
- Add documentation in example_application.rst for the 2 autosave request files.  There is already documentation in the source code in tomoscan.readPVFile().  Need to copy and improve this into .rst file.
- medm screens.  These need to be split up into a screen that is really beamline-independent and only has PVs from tomoScan.template, and a screen or screens that have the PVs from tomoScan_13BM.template.

**Comment by MarkRivers** [comment]
I have modified the medm screens as follows:
- tomoScan.adl and tomoScanEPICS_PVs.adl now only have the PVs in the base class, tomoScan.template.
- tomoScan.adl now takes a new BEAMLINE macro that specifies a beamline-specific adl file to load for other PVs.
- tomoScan_13BM.adl has been added.  It replaces the smaller screens with User, Config, and Sample information.  It is specific to 13-BM.  Other beamlines will need to make their versions of this screen, and pass that screen beam in the BEAMLINE macro.
- iocBoot/iocTomoScan/start_medm now passes the BEAMLINE=tomoScan_13BM.adl macro.
